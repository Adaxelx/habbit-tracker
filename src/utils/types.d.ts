export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: boolean;
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}
