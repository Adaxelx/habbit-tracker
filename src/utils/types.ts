export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
  handleDayChange: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: [
    {
      day: number;
      month: number;
      year: number;
      _id: string;
    },
  ];
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
  userId: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}

export type Color = { color?: string; id?: string };

export type NavProps = {
  header: string;
  moveDate: Function;
};

export interface DayCardProps {
  day: string;
  token: string | undefined;
}
