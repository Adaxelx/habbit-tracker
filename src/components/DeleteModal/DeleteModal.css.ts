import styled from 'styled-components';

interface DeleteModalProps {
  open: boolean;
}

export const StyledDeleteModal = styled.div<DeleteModalProps>`
  position: fixed;
  top: 10px;
  left: 50%;
  transform: translateX(-50%);
  width: 100%;
  max-width: 500px;
  padding: ${({ theme }) => theme.margin.s};
  background-color: white;
  font-size: ${({ theme }) => theme.font.sizes.sm};
  text-align: justify;
  border: 2px solid ${({ theme }) => theme.colors.border};
  z-index: 9998;
`;

interface ButtonProps {
  isDelete?: boolean;
}

export const StyledButton = styled.button<ButtonProps>`
  border: 1px solid ${({ theme }) => theme.colors.border};
  background-color: ${({ theme, isDelete }) =>
    isDelete ? theme.colors.error.main : theme.colors.success.main};
  padding: ${({ theme }) => theme.margin.xs};
  margin-right: ${({ theme }) => theme.margin.xs};
  &:hover {
    cursor: pointer;
  }
`;

export const StyledWrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-top: ${({ theme }) => theme.margin.xs};
`;
