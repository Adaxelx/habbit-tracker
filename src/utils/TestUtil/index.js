export { default } from './TestUtil.ts';
