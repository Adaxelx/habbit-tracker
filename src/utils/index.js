export { default as TestUtil } from './TestUtil';
export { default as RenderWithRouter } from './RenderWithRouter';
export * from './validators';
export * from './types.d';
