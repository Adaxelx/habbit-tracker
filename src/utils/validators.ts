export const createRequiredObject = (name: string) => ({
  required: `Field ${name} is required.`,
});

export const validatePattern = (pattern: RegExp) => ({
  pattern: {
    value: pattern,
    message: 'Incorrect characters.',
  },
});

export const emailValidation = validatePattern(
  // eslint-disable-next-line no-control-regex
  /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/,
);

export const createRestrictedLengthObject = (name: string) => ({
  ...createRequiredObject(name),
  minLength: {
    value: 6,
    message: `Field ${name} must have at least 6 characters.`,
  },
  maxLength: {
    value: 30,
    message: `Field ${name} must have maximum 30 characters.`,
  },
});

const timeValidFuntion = (timeLess: string | undefined, time: string | undefined) => {
  const message = 'Your time is not in boundaries.';
  if (timeLess && time) {
    const [hours, minutes] = time.split(':');
    const [hoursMin, minutesMin] = timeLess.split(':');
    if (hoursMin < hours) {
      return true;
    }
    if (hoursMin === hours) {
      if (minutesMin < minutes) {
        return true;
      }
      return message;
    }
    return message;
  }
  return message;
};

export const createTimeValidation = (min: string | undefined, max: string | undefined) => ({
  validate: {
    min: (value: string) => timeValidFuntion(min, value),
    max: (value: string) => timeValidFuntion(value, max),
  },
});
