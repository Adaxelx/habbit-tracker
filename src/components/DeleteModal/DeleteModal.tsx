import React, { MouseEventHandler } from 'react';
import { ClickOutside, Portal } from 'components';
import { StyledCloseButton, StyledLine } from 'components/PopUp/PopUp.css';
import { StyledDeleteModal, StyledButton, StyledWrapper } from './DeleteModal.css';

interface DeleteModalProps {
  open: boolean;
  handleClose: () => void;
  handleDelete: Function;
}

const DeleteModal = ({ open, handleClose, handleDelete }: DeleteModalProps) =>
  open ? (
    <Portal>
      <ClickOutside onOutsideClick={handleClose}>
        <StyledDeleteModal open={open}>
          <StyledCloseButton
            type="button"
            data-testid="deleteModal-close"
            onClick={handleClose as MouseEventHandler<HTMLButtonElement>}
          >
            <StyledLine rotate={45} />
            <StyledLine rotate={-45} />
          </StyledCloseButton>
          <p>Are you sure to delete this item? This action is not reversible.</p>
          <StyledWrapper>
            <StyledButton
              data-testid="delete-keep"
              onClick={handleClose as MouseEventHandler<HTMLButtonElement>}
            >
              Keep
            </StyledButton>
            <StyledButton
              isDelete
              data-testid="deleteItem"
              onClick={handleDelete as MouseEventHandler<HTMLButtonElement>}
            >
              Delete
            </StyledButton>
          </StyledWrapper>
        </StyledDeleteModal>
      </ClickOutside>
    </Portal>
  ) : null;

export default DeleteModal;
