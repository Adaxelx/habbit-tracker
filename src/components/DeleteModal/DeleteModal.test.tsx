import React from 'react';
import 'jest-styled-components';
import { TestUtil } from 'utils';
import DeleteModal from './DeleteModal';

describe('DeleteModal', () => {
  let util: TestUtil;
  let props;
  beforeEach(() => {
    jest.clearAllMocks();
    props = { open: true, handleClose: jest.fn(), handleOpen: jest.fn() };
    util = new TestUtil(<DeleteModal {...props} />);
  });

  it('should match snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });
});
