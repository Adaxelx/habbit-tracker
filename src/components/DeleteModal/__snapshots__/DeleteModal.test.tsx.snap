// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal should match snapshot 1`] = `
<DocumentFragment>
  .c2 {
  width: 50px;
  height: 50px;
  position: relative;
  display: block;
  margin-left: auto;
  margin-right: 0;
  border: none;
  background-color: transparent;
}

.c2:hover {
  cursor: pointer;
}

.c2:disabled {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
}

.c3 {
  height: 100%;
  width: 2px;
  position: absolute;
  top: 0%;
  left: 50%;
  background-color: #FF9191;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c4 {
  height: 100%;
  width: 2px;
  position: absolute;
  top: 0%;
  left: 50%;
  background-color: #FF9191;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c0 {
  cursor: pointer;
}

.c1 {
  position: fixed;
  top: 10px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 100%;
  max-width: 500px;
  padding: 0.75rem;
  background-color: white;
  font-size: 1.25rem;
  text-align: justify;
  border: 2px solid #707070;
  z-index: 9998;
}

.c6 {
  border: 1px solid #707070;
  background-color: #d9ffd9;
  padding: 0.5rem;
  margin-right: 0.5rem;
}

.c6:hover {
  cursor: pointer;
}

.c7 {
  border: 1px solid #707070;
  background-color: #FF9191;
  padding: 0.5rem;
  margin-right: 0.5rem;
}

.c7:hover {
  cursor: pointer;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 0.5rem;
}

<div
    class="c0"
    data-testid="clickOutside:wrapper"
  >
    <div
      class="c1"
      open=""
    >
      <button
        class="c2"
        data-testid="deleteModal-close"
        type="button"
      >
        <div
          class="c3"
          rotate="45"
        />
        <div
          class="c4"
          rotate="-45"
        />
      </button>
      <p>
        Are you sure to delete this item? This action is not reversible.
      </p>
      <div
        class="c5"
      >
        <button
          class="c6"
          data-testid="delete-keep"
        >
          Keep
        </button>
        <button
          class="c7"
          data-testid="deleteItem"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
